src/topp/PercolatorAdapter.cpp

```diff
@@ -638,6 +649,219 @@ class PercolatorAdapter :
+          }
+        }
+        if (string(sqlite3_column_name( stmt, i )) == "DECOY")
+        {
+          if(sqlite3_column_int( stmt, i ) == 1)
+          {
+            label = -1;
+          }
+          else
+          {
+            label = 1;
+          }
+        }
+        if (string(sqlite3_column_name( stmt, i )) == "MODIFIED_SEQUENCE")
+        {
+          peptide = std::string(reinterpret_cast<const char*>(sqlite3_column_text( stmt, i )));
+        }
+        if (string(sqlite3_column_name( stmt, i )).substr(0,4) == "VAR_")
+        {
+          features[string(sqlite3_column_name( stmt, i ))] = sqlite3_column_double( stmt, i );
+        }
+      }
+
+      // Write output
+      if (k == 0)
+      {
+        pin_output << "PSMId\tLabel\tScanNr";
+        for(auto const &feat : features)
+        {
+          pin_output << "\t" << feat.first;
+        }
+        pin_output << "\tPeptide\tProteins\n";
+      }
+      pin_output << psm_id << "\t" << label << "\t" << scan_id;
+      for(auto const &feat : features)
+      {
+        pin_output << "\t" << feat.second;
+      }
+      pin_output << "\t." << peptide << ".\tProt1" << "\n";
+
+      sqlite3_step( stmt );
+      k++;
+    }
+
+    sqlite3_finalize(stmt);
+    sqlite3_close(db);
+
+    return EXECUTION_OK;
+  }
+
+  static int callback(void * /* NotUsed */, int argc, char **argv, char **azColName){
+    int i;
+    for(i=0; i<argc; i++)
+    {
+      printf("%s = %s\n", azColName[i], argv[i] ? argv[i] : "NULL");
+    }
+    printf("\n");
+    return(0);
+  }
+
+  void writeOSWFile_(std::string in_osw, std::string osw_level, std::map<String, PercolatorResult> pep_map) {
+    std::string table;
+    std::string create_sql;
+
+    if (osw_level == "ms1") {
+      table = "SCORE_MS1";
+      create_sql =  "DROP TABLE IF EXISTS " + table + "; " \
+                    "CREATE TABLE " + table + "(" \
+                    "FEATURE_ID INT NOT NULL," \
+                    "SCORE DOUBLE NOT NULL," \
+                    "QVALUE DOUBLE NOT NULL," \
+                    "PEP DOUBLE NOT NULL);";
+
+    } else if (osw_level == "transition") {
+      table = "SCORE_TRANSITION";
+      create_sql =  "DROP TABLE IF EXISTS " + table + "; " \
+                    "CREATE TABLE " + table + "(" \
+                    "FEATURE_ID INT NOT NULL," \
+                    "TRANSITION_ID INT NOT NULL," \
+                    "SCORE DOUBLE NOT NULL," \
+                    "QVALUE DOUBLE NOT NULL," \
+                    "PEP DOUBLE NOT NULL);";
+
+    } else {
+      table = "SCORE_MS2";
+      create_sql =  "DROP TABLE IF EXISTS " + table + "; " \
+                    "CREATE TABLE " + table + "(" \
+                    "FEATURE_ID INT NOT NULL," \
+                    "SCORE DOUBLE NOT NULL," \
+                    "QVALUE DOUBLE NOT NULL," \
+                    "PEP DOUBLE NOT NULL);";
```

Comment:
do you add these tables? again, it seems like a good idea to keep all of the code in the `FORMAT` folder so that we have some overview over which tables (may) exist and how to read these tables

Reply:
Good point. I will move the functions to ``FORMAT``.